lib/std/special/compiler_rt/ctzdi2_test.zig

```diff
@@ -0,0 +1,329 @@
+const ctz = @import("count0bits.zig");
+const testing = @import("std").testing;
+
+fn test__ctzdi2(a: u64, expected: i32) !void {
+    // XXX At high optimization levels this test may be horribly miscompiled if
+    // one of the naked implementations is selected.
+    var nakedCtzdi2 = ctz.__ctzdi2;
+    var actualCtzdi2 = @ptrCast(fn (a: i64) callconv(.C) i32, nakedCtzdi2);
+    var x = @bitCast(i64, a);
+    var result = actualCtzdi2(x);
+    try testing.expectEqual(expected, result);
```

Comment:
what's going on here? I don't understand why this has to be complicated, can't it just call the function that it's testing like normal?

Reply:
The idea was to provide native assembly eventually, ie [the ones for x86](https://en.wikipedia.org/wiki/X86_Bit_manipulation_instruction_set#BMI1_(Bit_Manipulation_Instruction_Set_1) since we dont have a peephole optimizer without LLVM.

Would you prefer leaving this out (for now)?

Reply:
Yes please. Let's not have this XXX naked stuff and TODO comments everywhere when nothing is actually wrong. If there should be an inline assembly version of a function, there can be a github issue for that.

Related: #363

Reply:
I removed that stuff in YAGNI commits.